### PR TITLE
Fix root cause of unit tests failing randomly

### PR DESCRIPTION
**Pull request by MatkovIvan**
* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] Did you add unit tests?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Make sure that tests are waiting for background operations

## Related PRs or issues

[AB#59839](https://msmobilecenter.visualstudio.com/web/wi.aspx?pcguid=27bf89de-9697-418a-accf-1aa125b22914&id=59839)

### CODE REVIEW

**Comment by MatkovIvan** [comment]
CI passed. Retry

**Comment by MatkovIvan** [comment]
`testDeviceExtJSONSerializingToDictionary`, Test crashed

🤔

**Comment by MatkovIvan** [comment]
CI passed. Retry for sure

**Comment by MatkovIvan** [comment]
CI passed x2. Retry again

File: AppCenter/AppCenterTests/MSAppCenterTests.m [hunk 1]
```diff
@@ -46,22 +50,32 @@ @implementation MSAppCenterTest
...
+
+  // Session context.
   [MSSessionContext resetSharedInstance];
   self.sessionContextMock = OCMClassMock([MSSessionContext class]);
   OCMStub([self.sessionContextMock sharedInstance]).andReturn(self.sessionContextMock);
 }
 
 - (void)tearDown {
   [self.settingsMock stopMocking];
+  [self.channelGroupMock stopMocking];
```
Comment (clpolet) [inline AppCenter/AppCenterTests/MSAppCenterTests.m]:
```suggestion
  [self.channelGroupMock stopMocking];
  [self.channelUnitMock stopMocking];
```

Comment (MatkovIvan) [reply AppCenter/AppCenterTests/MSAppCenterTests.m]: No, we shouldn't do it for `OCMProtocolMock`

File: AppCenter/AppCenterTests/MSChannelGroupDefaultTests.m [hunk 2]
```diff
@@ -76,51 +84,44 @@ - (void)testAddNewChannel {
...
+  XCTAssertEqual(self.ingestionMock, channelUnit.ingestion);
 }
 
 - (void)testAddChannelWithCustomIngestion {
 
   // If
-  id<MSIngestionProtocol> ingestionMockDefault = OCMPartialMock([MSAppCenterIngestion new]);
-  id<MSIngestionProtocol> ingestionMockCustom = OCMPartialMock([MSAppCenterIngestion new]);
-  MSChannelGroupDefault *sut = [[MSChannelGroupDefault alloc] initWithIngestion:ingestionMockDefault];
+  id<MSIngestionProtocol> ingestionMockCustom = OCMClassMock([MSAppCenterIngestion class]);
```
Comment (guperrot) [inline AppCenter/AppCenterTests/MSChannelGroupDefaultTests.m]: It just failed on last build with that particular test that was updated.

**Comment by guperrot** [comment]
Also please do not retry PR build if the build fails in MSAuthTokenContext or anything related to channel or delegates. To retry you need to make changes to the test code.

**Comment by clpolet** [comment]
Cancelled the CI after tests passed on all platforms. Triggering it again to see if we just got lucky.

File: AppCenter/AppCenterTests/MSChannelGroupDefaultTests.m [hunk 3]
```diff
@@ -39,18 +39,12 @@ - (void)setUp {
...
+  self.sut.logsDispatchQueue = dispatch_get_main_queue();
 }
 
 - (void)tearDown {
   [super tearDown];
-
-  // Wait all tasks in tests.
-  XCTestExpectation *expectation = [self expectationWithDescription:@"tearDown"];
-  dispatch_async(self.sut.logsDispatchQueue, ^{
-    dispatch_suspend(self.sut.logsDispatchQueue);
```
Comment (MatkovIvan) [inline AppCenter/AppCenterTests/MSChannelGroupDefaultTests.m]: @clpolet 
Disagree with this change. It fixes the "chain actions" case
Restored with comments in #1491

**Pull request merged**